src/GafferScene/RenderManifest.cpp

```diff
@@ -0,0 +1,445 @@
+	{
+		// Not sure it's worth doing a full text compare, but if the pointers compare equal, that's a quick
+		// and conservative test that the manifest stored in the metadata hasn't changed.
+		if( cryptoManifestStringData == m_loadedFromCryptomatteJSONStringData )
+		{
+			return;
+		}
+
+		const std::string &cryptoManifestString = cryptoManifestStringData->readable();
+		boost::iostreams::stream<boost::iostreams::array_source> stream( cryptoManifestString.c_str(), cryptoManifestString.size() );
+
+		loadCryptomatteJSON( stream );
+
+		m_loadedFromFilePath = "";
+		m_loadedFromCryptomatteJSONStringData = cryptoManifestStringData;
+
+		return;
+
+	}
+
+	std::filesystem::file_time_type currentModTime;
+	try
+	{
+		currentModTime = std::filesystem::last_write_time( sideCarManifestPath );
+	}
+	catch( std::exception &e )
+	{
+		throw IECore::Exception( std::string( "Could not find manifest file : " ) + sideCarManifestPath + " : " + e.what() );
+	}
+
+	if( sideCarManifestPath == m_loadedFromFilePath && currentModTime == m_loadedFromFilePathModTime )
+	{
+		// We're using a manifest file, and the timestamp hasn't changed since we were last loaded from this file
+		return;
+	}
+
+	if( !isCryptomatte )
+	{
+		loadEXRManifest( sideCarManifestPath.c_str() );
+	}
+	else
+	{
+		std::ifstream stream( sideCarManifestPath.c_str() );
+		loadCryptomatteJSON( stream );
+	}
+
+	m_loadedFromFilePath = sideCarManifestPath;
+	m_loadedFromFilePathModTime = currentModTime;
+	m_loadedFromCryptomatteJSONStringData.reset();
+}
+
+void RenderManifest::loadEXRManifest( const std::filesystem::path &filePath )
+{
+	m_map.clear();
+	modified();
+
+	Imf::MultiPartInputFile exrInput( filePath.generic_string().c_str() );
+	int idManifestPart = -1;
+	for( int part = 0; part < exrInput.parts (); part++ )
+	{
+		if( Imf::hasIDManifest( exrInput.header( part ) ) )
+		{
+			idManifestPart = part;
+			break;
+		}
+	}
+
+	if( idManifestPart == -1 )
+	{
+		throw IECore::Exception( "No manifest found" );
+	}
+
+	const Imf::Header header = exrInput.header( idManifestPart );
+
+	const Imf::CompressedIDManifest& compressedManifest = Imf::idManifest( header );
+	const Imf::IDManifest manifest( compressedManifest );
+	const Imf::IDManifest::ChannelGroupManifest &idManifest = manifest[0];
+
+	for( Imf::IDManifest::ChannelGroupManifest::ConstIterator i = idManifest.begin(); i != idManifest.end(); ++i )
+	{
+		m_map.insert( PathAndID( ScenePlug::stringToPath( i.text()[0] ), i.id() ) );
+	}
+}
+
+void RenderManifest::writeEXRManifest( const std::filesystem::path &filePath ) const
+{
+	if( !boost::ends_with( filePath.generic_string(), ".exr" ) )
+	{
+		throw IECore::Exception( "ID manifest file path does not end in \".exr\": " + filePath.generic_string() );
+	}
+
+	if( !filePath.is_absolute() )
+	{
+		throw IECore::Exception( "ID manifest file path is not an absolute file path : " + filePath.generic_string() );
+	}
+
+	std::filesystem::create_directories( filePath.parent_path() );
```

Comment:
I think this belongs one level up, in Render and RenderController.

Reply:
Makes sense. While fixing this, I decided to include 197c04766767c8d9ead5d7a12a4bc8f177701f0f in the RenderManifest PR - it's not technically in the range of commits you requested, but it's a simple change, and the matching change is Render is included, so it makes sense to include it as well.